src/staker.ts

```diff
@@ -0,0 +1,143 @@
+import { BigintIsh, Token, validateAndParseAddress } from '@uniswap/sdk-core'
+import { MethodParameters, toHex } from './utils/calldata'
+import { Interface } from '@ethersproject/abi'
+import { abi } from '@uniswap/v3-staker/artifacts/contracts/UniswapV3Staker.sol/UniswapV3Staker.json'
+import { Pool } from './entities'
+
+/**
+ * Represents a unisque staking program.
+ */
+export interface IncentiveKey {
+  /**
+   * The token awarded in the program.
+   */
+  token: Token
+  /**
+   * The pool that the staked positions must provide in.
+   */
+  pool: Pool
+  /**
+   * The time when the incentive program begins.
+   */
+  startTime: BigintIsh
+  /**
+   * The time that the incentive program ends.
+   */
+  endTime: BigintIsh
+  /**
+   * The address which receives any remaining reward tokens at `endTime`.
+   */
+  refundee: string
+}
+
+/**
+ * Options to specify when claiming rewards.
+ */
+export interface ClaimOptions {
+  /**
+   * The id of the NFT
+   */
+  tokenId: BigintIsh
+
+  /**
+   * The token rewarded for participating in the staking program.
+   */
+  rewardToken: string
+
+  /**
+   * Address to send rewards to.
+   */
+  recipient: string
+
+  /**
+   * The amount of `rewardToken` to claim.
+   */
+  amount: BigintIsh
+
+  /**
+   * The amount of `rewardToken` to claim.
+   */
+  withdraw?: boolean
+
+  /**
+   * The owner of the position.
+   */
+  owner?: string
+
+  /**
+   * Optional parameter passed to the callback `onERC721Received`.
+   */
+  data?: string
+}
+
+export abstract class Staker {
+  public static INTERFACE: Interface = new Interface(abi)
+
+  protected constructor() {}
+
+  /** To claim rewards, must unstake and then claim. */
```

Comment:
let's use jsdoc here. i suggest using a code editor that mostly fills in the jsdoc for you once you type `/**` and hit enter like intellij